File: datetime/easter/easter.py
import datetime

def get_easter_for(year):
    g = (year % 19) + 1
    c = (year // 100) + 1
    x = (3 * c // 4) - 12
    z = (((8 * c) + 5) // 25) - 5
    d = (5 * year // 4) - x - 10
    e = ((11 * g) + 20 + z - x) % 30
    if ((e == 25 and g > 11) or e == 24):
        e += 1
    n = 44 - e
    if n < 21:
        n += 30
    n += 7 - ((d + n) % 7)

    if n > 31:
        return datetime.datetime(year, 4, n - 31)
    return datetime.datetime(year, 3, n)

def generate_easters(start_year, end_year):
    for year in range(start_year, end_year + 1):
        yield get_easter_for(year)

def print_easters(easters):
    print("Easters:")
    for easter in easters:
        print(easter.strftime("   %d %B, %Y"))

print_easters(generate_easters(1950, 2050))

File: datetime/easter/easter.r
library(coro)

get_easter_for <- function(year) {
  g <- (year %% 19) + 1
  c <- floor(year / 100) + 1
  x <- floor(3 * c / 4) - 12
  z <- floor(((8 * c) + 5) / 25) - 5
  d <- floor(5 * year / 4) - x - 10
  e <- ((11 * g) + 20 + z - x) %% 30
  if ((e == 25 && g > 11) || e == 24) {
    e <- e + 1
  }
  n <- 44 - e
  if (n < 21) {
    n <- n + 30
  }
  n <- n + 7 - ((d + n) %% 7)

  if (n > 31) {
    ISOdate(year, month = 4, day = n - 31)
  } else {
    ISOdate(year, month = 3, day = n)
  }
}

generate_easters <- generator(function(start_year, end_year) {
  for (year in start_year:end_year) {
    yield(get_easter_for(year))
  }
})

print_easters <- function(easters) {
  print("Easters:")
  loop(for (easter in easters) {
    print(format(easter, format = "   %d %B, %Y"))
  })
}

print_easters(generate_easters(1950, 2050))
